packages/apps/src/microsoft/teams/apps/options.py

```diff
@@ -5,27 +5,35 @@
 
 from dataclasses import dataclass, field
 from logging import Logger
-from typing import Any, List, Optional
+from typing import Any, List, Optional, TypedDict
 
 from microsoft.teams.common.storage import Storage
 
 from .plugins import PluginBase
 
 
-@dataclass
-class AppOptions:
+class AppOptions(TypedDict, total=False):
```

Comment:
by setting total=False, does it still check that all req fields are being passed in?

Reply:
No it doesn't (which is what we want, since everything is optional)
> This indicates that all keys defined within the TypedDict are optional by default. Instances of the TypedDict can be created without including all, or even any, of the declared keys, and a type checker will not raise an error for missing keys.

Reply:
But i switched to using NotRequired for now. I think it's cleaner